### DISCUSSION TITLE
Template Serial Battery

### DISCUSSION THREAD

**Discussion by RFDarter** [Louisvdw/dbus-serialbattery#980] [Ideas]
It would be nice if we could have a template serial battery that will not directly connect to an bms but instead via USB to a esp32 or similar.

That way we could aggregate multiple batterys and implement a charge logic on the esp32 instead of the config file in the venus os.

Like we can do with the mqtt-battery
https://github.com/mr-manuel/venus-os_dbus-mqtt-battery

**Comment by mr-manuel** [comment]
Feel free to develop an ESP32 software and then write a battery driver for this driver. In my eyes a lot of work for something that already works with dbus-serialbattery. No ESP32 needed.

**Comment by RFDarter** [comment]
Doing it with esphome it has a lot more Potential than doing it in a config file on the venus os.

Ease of use wozld be the big one.
Just messing with the config on the venus os once and to the rest via a web interface.

Changing the current limit and discharge current limit via a mobile phone on a Web interface.

Possibility to include it into home assistant and automating the limits.

Just to name a few.

Would you be willing to merge my battery driver if I write one?

**Reply by mr-manuel** [reply]
Sure, if you provide the whole documentation end needed ESP files. Then it's a driver like `bms/esphome.py`.

**Reply by mr-manuel** [reply]
Check also https://louisvdw.github.io/dbus-serialbattery/general/supported-bms#add-by-opening-a-pull-request
